src/arm/isa-arm.c

```diff
@@ -576,6 +584,14 @@ DEFINE_MULTIPLY_INSTRUCTION_XY_ARM(SMLA,
 
 DEFINE_MULTIPLY_INSTRUCTION_XY_ARM(SMUL, cpu->gprs[rd] = x * y;)
 
+DEFINE_MULTIPLY_INSTRUCTION_WY_ARM(SMLAW,
+	int32_t dn = cpu->gprs[rn]; \
+	int64_t d = (((int64_t)cpu->gprs[rm]) * ((int64_t)y)) >> 16; \
+	cpu->gprs[rd] = d + dn; \
+	cpu->cpsr.q = cpu->cpsr.q || ARM_V_ADDITION(d, dn, cpu->gprs[rd]);)
```

Comment:
ARM_V_ADDITION is for determining if two 32-bit additions overflow without converting them to 64 bit. Since it's already 64-bit here, not only will ARM_V_ADDITION not work, you can just do `d + dn > 0xFFFFFFFFLL`.